ERCS/erc-7722.md

```diff
@@ -0,0 +1,328 @@
+
+Example with reorg and null-value basket transfers:
+```
+A owns basket-a1{..., value:10}
+B owns basket-b1{..., value:5}, basket-b2{..., value:15}, ...
+A: transfer basket-a1 to B
+B: reorg [basket-a1, basket-b1, basket-b2]
+      to [basket-b3{..., value:10}, basket-b4{..., value:10}, basket-b5:{..., value:10},
+            basket-b6:{..., value:0}, basket-b7:{..., value:0}]
+      where sum of inputs is the sum of outputs
+B: transfer basket-b5{value:10} to C
+B: transfer basket-b6{value:0}  to D
+B: transfer basket-b7{value:0}  to E
+```
+
+If B would directly send basket-a1 to C, A would know what C is receiving, however, now that B has reorg'ed the baskets, A can not know anymore what has been sent to C.
+
+Moreover, observers still see who is communicating with whom, but since there is noise introduced, they can not tell which of these transfers are actually transferring real values.
+
+## Rationale
+
+### Breaking the ERC-20 Compatibility
+
+The transparency inherent in ERC-20 tokens presents a significant issue for reusable blockchain identities, such as defined by ERC-725. To address this, we prioritize privacy over ERC-20 compatibility, ensuring the confidentiality of token balances.
+
+### Reorg Oracles
+
+The trusted oracles and the minimum number of required signatures can be configured to achieve the desired level of decentralization.
+
+The basket holder proposes the input and output baskets for the reorg, while the oracles are responsible for verifying that the sums of the values on both sides (input and output) are equal. This system allows for mutual control, ensuring that no single party can manipulate the process.
+
+Fraudulent oracles can be tracked back on-chain, i.e., the system ensures weak-integrity at minimum.
+
+To further strengthen the integrity, it would also be possible to apply Zero-Knowledge Proofs (ZKP) to provide reorg proofs, however, we have chosen to use oracles for efficiency and simplicity reasons.
+
+### Off-chain Data Storage
+
+We have chosen the token operator, which in most cases will be the issuer or registrar, as the initial and main source for off-chain data. This is acceptable, since they must know anyway which investor holds which positions to manage lifecycle events on the token. While this approach may not be suitable for every use case within the broader Ethereum ecosystem, it fits well the financial instruments in the regulated environment of the financial industry, which rely on strict KYC and token operation procedures.
+
+## Backwards Compatibility
+
+* Opaque Token is not compatible with ERC-20 for reasons explained in the Rationale section.
+
+## Reference Implementation
+
+To be included as inline code or in `../assets/eip-####/`.
+
+## Security Considerations
+
+### Fraudulent Oracles
+To be discussed...
+
+### Oracles Collecting Confidential Data
+To be discussed...
+
+### Confidential Data Loss
+To be discussed...
```

Comment:
```suggestion
<!-- TODO -->
```

The linter will make sure you replace HTML-style comments before you get too far in the process.